doc/source/contributing/syntax_guide.md

```diff
@@ -0,0 +1,236 @@
+(syntax_guide)=
+# Written content formatting reference
+
+## Adding targets
+Adding custom targets or anchors to the headings is really helpful for cross-referencing. They allow us to link to the heading using a simple syntax.
+They are defined using this syntax:
+
+::::{tab-set}
+
+:::{tab-item} rST
+:sync: rst
+
+```
+.. _mytarget:
+```
+:::
+
+:::{tab-item} MyST (Markdown)
+:sync: myst
+
+```
+(mytarget)=
+```
+:::
+
+::::
+
+They are referred using this syntax:
+
+::::{tab-set}
+
+:::{tab-item} rST
+:sync: rst
+
+```
+:ref:`mytarget`
+```
+:::
+
+:::{tab-item} MyST (Markdown)
+:sync: myst
+
+```
+{ref}`mytarget`
+```
+:::
+
+::::
+
+For adding anchors in `.ipynb` files, Markdown syntax will be used in the markdown cells of `.ipynb` files.
+
+:::{seealso}
+{ref}`sphinx:ref-role`
+  Sphinx documentation about the `ref` role.
+
+[Hyperlink targets](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#hyperlink-targets)
+  Docutils documentation on targets (another name for anchors) with extensive detail on how and when
+  they can be used.
+
+[MyST targets and cross-referencing docs](https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html#targets-and-cross-referencing)
+:::
+
+## Backticks for highlighting code keywords
+
+For highlighting inline code keywords or file names, backticks are used. In Markdown single backticks are used while in rST double backticks are used. For example, for highlighting the file name `conf.py`, we will use this syntax:
+
+:::::{tab-set}
+::::{tab-item} rST
+```
+``conf.py``
+```
+::::
+::::{tab-item} MyST (Markdown)
+```
+`conf.py`
+```
+::::
+:::::
+
+## Table of content tree
+All documentation generated with sphinx is structured via `toctrees`.
+Sphinx prints a warning if a file is not part of any toctree unless
+that file is marked as an orphan.
+
+The hierarchy defined via toctrees (and not the file hierarchy!) is
+the one that defines the nabvar and the contents of the left sidebar.
+Keeping the toctrees organized and up to date ensures the sphinx
+build works, that the generated documentation is correctly ordered
+and can be navigated and that all pages can be reached.
+It also enables navigation to the **Previous** and **Next** pages in the footer.
+
+Follow this syntax for adding the table of content:
+
+:::::{tab-set}
+::::{tab-item} rST
+```rST
+.. toctree::
+    developer_guide
+    doc_guide
+```
+::::
+::::{tab-item} MyST (Markdown)
+```markdown
+:::{toctree}
+developer_guide
+doc_guide
+:::
+```
+::::
+:::::
+
+Read more about the {ref}`Sphinx toctree directive <sphinx:table-of-contents>`.
+
+(adding_references)=
+## Adding references
+
+In ArviZ docs, we use {confval}`sphinx:intersphinx_mapping` to add references to other libraies functions and objects.
```

Comment:
```suggestion
In ArviZ docs, we use {confval}`sphinx:intersphinx_mapping` to add references to other libraries functions and objects.
```
The link preview is supposed to be like this?

Reply:
no it should not. It looks like a bug in myst.

Reply:
fixed, turns out confval cross-references are not included in sphinx, they must be added manually if we want to reference them.

Reply:
yes, it is working fine now.